spec/kittikrb/cursor_spec.rb

```diff
@@ -122,15 +122,82 @@ def current_rendered_buffer
       end
     end
 
-    describe '#move_to' do
+    context 'move methods' do
       def current_coords
         [cursor.instance_variable_get(:@x), cursor.instance_variable_get(:@y)]
       end
 
-      it 'should properly set absolute position of cursor' do
+      before do
         expect(current_coords).to eq [0, 0]
-        expect(cursor.move_to(5, 10)).to be_instance_of described_class
-        expect(current_coords).to eq [5, 10]
+      end
+
+      describe '#move_to' do
+        it 'should properly set absolute position of cursor' do
+          expect(cursor.move_to(5, 10)).to be_instance_of described_class
+          expect(current_coords).to eq [5, 10]
+        end
+      end
+
+      describe '#move_by' do
+        it 'should properly move cursor right and down' do
+          expect(cursor).to receive(:right).with(7)
+          expect(cursor).to receive(:down).with(3)
+          expect(cursor.move_by(7, 3)).to be_instance_of described_class
```

Comment:
Ok, I understand your problem here. We should check side effect of our stubbed methods `right` and `down`. Simply, we can just write two tests. We already have one of them. So you should write another one which will check `current_cords`. 

```
        it 'should change current coordinates' do
          cursor.move_by(7, 3)
          expect(current_coords).to eq [7, 3]
          cursor.move_by(-1, -1)
          expect(current_coords).to eq [6, 2]
        end
```

I'm sure, it is not better solution, but let's refactor it later.